src/main/java/com/google/sps/data/GivrUser.java

```
@@ -51,48 +53,118 @@ public boolean isMaintainer() {
-      
-    FetchOptions fetchOptions = FetchOptions.Builder.withLimit(1);
+
     PreparedQuery preparedQuery = datastore.prepare(query);
+    return preparedQuery;
+  }
+
+  // Check if User with propertyName, propertyValue exists within Datastore.
+  public static boolean checkIfUserWithPropertyExists(String propertyName, String propertyValue) {
+    PreparedQuery preparedQuery = getPreparedQueryResultUserWithProperty(propertyName, propertyValue);
+    FetchOptions fetchOptions = FetchOptions.Builder.withLimit(1);
+    QueryResultList<Entity> userResult = preparedQuery.asQueryResultList(fetchOptions);
+
+    if (userResult.size() < 1) {
+      return false;
+    } else if (userResult.size() > 1) {
+      throw new IllegalArgumentException("More than one user with the " + propertyName + " was found.");
+    }
+    return true;
+  }
+
+  // Updates a User entity in Datastore, identifying with the first two parameters with values from second two parameters.
+  public static void updateUserInDatastore(String identifyingProperty, String identifyingValue, String updateProperty, String updateValue) {  
+    if (!checkIfUserWithPropertyExists(identifyingProperty, identifyingValue)) {
+      throw new IllegalArgumentException("User with " + identifyingProperty + " of value: " + identifyingValue + " was not found in the Datastore.");
+    }
+
+    PreparedQuery preparedQuery = getPreparedQueryResultUserWithProperty(identifyingProperty, identifyingValue);
+    FetchOptions fetchOptions = FetchOptions.Builder.withLimit(1);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    for (Entity entity: preparedQuery.asIterable(fetchOptions)) {
+      Transaction txn = datastore.beginTransaction();
+      try {
+        entity.setProperty(updateProperty, updateValue);
+        datastore.put(txn, entity);
+
+        txn.commit();
+      } finally {
+        if (txn.isActive()) {
+          txn.rollback();
```

should you throw an exception here to tell the caller that the update failed?